.release-notes/3650.md

```diff
@@ -0,0 +1,17 @@
+## Graceful error handling on `consume` expressions
```

Comment:
I think what is missing from these release notes is an that previously some of these things might have been compiler errors etc before. The fix ensures some things, but what does that mean for the end-user?

Something along the lines of "previously any of X would result in Y that has now been fixed" is an important part of the release notes messaging.

Reply:
Rephrased. Can you please check if this sounds better? also I removed one of the points. I notice that 0.3.3 already covered it. (That particular fix was only refactored this time around).